scripts/deploy.sh

```diff
@@ -1,49 +1,86 @@
 #!/bin/bash
 
-#Creación de la carpeta tcph y dentro descompresión el benchmark
-mkdir tpch
-
-##Instalación de la herramienta zip si no está.
-echo Instalando herramienta "zip"
-sudo apt-get install zip
-
-mv tpch.zip ./tpch
-cd tpch/
-unzip tpch.zip
-cd tpch_2_17_0/dbgen/
-#mv makefile.suite makefile
-
-### cambiar las lineas del makefile para que se adapten a la máquina.
-## En realidad todo esto podría sustituirse por copiar el makefile ya modificado.
-
-echo Instalando herramienta "gcc"
-sudo apt-get install gcc
-
-echo Instalando herramienta "make"
-sudo apt-get install make
+##
+##  Developed by Scenic
+##
+
+# variables
+
+DIRECTORY=cloudabee
+ZIP=tpch.zip
+DIRECTORY_TPCH=tpch_2_17_0/dbgen/
+
+# subprograms
+function advertement {
+  echo "CAUTION: TPC-H is licensed and we can NOT distribute it.";
+  echo "   You should download it from http://www.tpc.org/tpc_documents_current_versions/current_specifications.asp";
+  echo "   In the following, we assume there is a file \"$ZIP\" in the directory where deploy.sh is called.";
+  echo "";
+}
+
+function checking_tool {
+  if ! hash $1 2>/dev/null; then
+    echo "Tool $1 needs to be installed";
+    sudo apt-get install $1;
+  fi
+}
+
+# script
+
+advertement
+
+echo "1. Checking if needed tools are available in the system.";
+checking_tool zip
+checking_tool gcc
+checking_tool make
+
+if [ -d $DIRECTORY ]; then
+  echo "ERROR: $DIRECTORY exists. You should delete it to continue.";
+  exit;
+fi
+if [ ! -f $ZIP ]; then
+  echo "ERROR: $ZIP does NOT exist. You should download it to continue.";
+  exit;
+fi
+
+mkdir $DIRECTORY
+cp $ZIP $DIRECTORY
+cd $DIRECTORY
+
+echo "2. Unziping $ZIP.";
+unzip -qq $ZIP
+
+if [ ! -d $DIRECTORY_TPCH ]; then
+  echo "ERROR: $DIRECTORY_TPCH does not exist. You should download the correct version of TPC-H to continue.";
+  exit;
+fi
+
+cd $DIRECTORY_TPCH
+
+echo "3. Compiling.";
 make
 
-##Se crea una bd de un GB.
+###Se crea una bd de un GB.
 ./dbgen -s 1
-
+#
 mkdir DATA
 mv *.tbl DATA
 sudo mv DATA /var/lib/mysql/
 
 ##Instalación de mysql.
-#sudo debconf-set-selections <<< 'mysql-server mysql-server/root_password password jose123'
-#sudo debconf-set-selections <<< 'mysql-server mysql-server/root_password_again password jose123'
+sudo debconf-set-selections <<< 'mysql-server mysql-server/root_password password jose123'
+sudo debconf-set-selections <<< 'mysql-server mysql-server/root_password_again password jose123'
 sudo apt-get -y install mysql-server
```

Comment:
Great job! I think that mysql must be installed previously to  move DATA folder to /var/lib/mysql/.